plugins/pay.c

```diff
@@ -491,6 +514,8 @@ static struct command_result *listsendpays_done(struct command *cmd,
 			pm->sortkey.payment_hash = pm->payment_hash;
 			pm->sortkey.groupid = groupid;
 			pm->success_at = UINT64_MAX;
+			pm->created_index = created_index;
+			pm->updated_index = updated_index;
```

Comment:
OK, here's the problem the other side of this "if (!pm)" statement.  If we have seen part of this payment before, what do we use the created and update indexes?  

For updated_index, I *think* we have to use the greatest we've seen (if any part of a payment gets updated, it's been updated), and for created_index I think it's the smallest we've seen (that's stable: we might create a new part of an existing payment).

This makes sense if we were to implement "wait" for pays, too (that requires more infrastructure to allow that to work in plugins).

Reply:
I'm sorry i don't have a clue what's going on here, i just put it there because i saw the other values there :/




> For updated_index, I _think_ we have to use the greatest we've seen (if any part of a payment gets updated, it's been updated), and for created_index I think it's the smallest we've seen (that's stable: we might create a new part of an existing payment).

Yes, but i don't know how to do it, sorry.

Reply:
No problems, I can fix this up for you to review!